src/highlighters.cc

```diff
@@ -563,7 +574,12 @@ UniquePtr<Highlighter> create_column_highlighter(HighlighterParameters params, H
                         atom_it = ++line.split(atom_it, remaining_col);
                     if (atom_it->length() > 1)
                         atom_it = line.split(atom_it, 1_col);
-                    atom_it->face = merge_faces(atom_it->face, face);
+                    if (!ruler_provided)
+                        atom_it->face = merge_faces(atom_it->face, face);
+                    if (ruler_provided && is_blank(atom_it->content()[0])) {
```

Comment:
Kakoune codebase uses the alternate spelling for boolean operator, `and` here rather than `&&`, and curly brace on its own line. Moreover you probably can make this an `else if` to avoid testing `ruler_provided` again.